lnrpc/marshall_utils.go

```diff
@@ -179,3 +180,67 @@ func MarshalOutPoint(op *wire.OutPoint) *OutPoint {
 		OutputIndex: op.Index,
 	}
 }
+
+// ParseCoinSelectionStrategyObject parses a wallet.CoinSelectionStrategy
+// object to its string counterpart.
+func ParseCoinSelectionStrategyObject(strategy wallet.CoinSelectionStrategy) (
+	*string, error) {
+
+	switch strategy {
+	case wallet.CoinSelectionLargest:
+		coinSelectionStrategy := "largest"
+		return &coinSelectionStrategy, nil
+	case wallet.CoinSelectionRandom:
+		coinSelectionStrategy := "random"
+		return &coinSelectionStrategy, nil
+	default:
+		return nil, fmt.Errorf("unknown coin selection strategy "+
+			"%v", strategy)
+	}
+}
+
+// ParseCoinSelectionStrategy parses a coin selection strategy string
+// from the CLI to its lnrpc.CoinSelectionStrategy counterpart proto type.
+func ParseCoinSelectionStrategy(strategy string) (
+	*CoinSelectionStrategy, error) {
+
+	switch strategy {
+	case "global-config":
+		strategyVal := CoinSelectionStrategy_STRATEGY_USE_GLOBAL_CONFIG
```

Comment:
Okay what should we return in case of no match ? `nil` gives an error since `CoinSelectionStrategy` is a proto type.
```
// ParseCoinSelectionStrategy parses a coin selection strategy string
// from the CLI to its lnrpc.CoinSelectionStrategy counterpart proto type.
func ParseCoinSelectionStrategy(strategy string) (
	CoinSelectionStrategy, error) {

	switch strategy {
	case "global-config":
		return CoinSelectionStrategy_STRATEGY_USE_GLOBAL_CONFIG, nil

	case "largest":
		return CoinSelectionStrategy_STRATEGY_LARGEST, nil

	case "random":
		return CoinSelectionStrategy_STRATEGY_RANDOM, nil

	default:
		return nil, fmt.Errorf("unknown coin selection strategy "+
			"%v", strategy)
	}
}
```Okay what should we return in case of no match ? `nil` gives an error since `CoinSelectionStrategy` is a proto type.
```
// ParseCoinSelectionStrategy parses a coin selection strategy string
// from the CLI to its lnrpc.CoinSelectionStrategy counterpart proto type.
func ParseCoinSelectionStrategy(strategy string) (
	CoinSelectionStrategy, error) {

	switch strategy {
	case "global-config":
		return CoinSelectionStrategy_STRATEGY_USE_GLOBAL_CONFIG, nil

	case "largest":
		return CoinSelectionStrategy_STRATEGY_LARGEST, nil

	case "random":
		return CoinSelectionStrategy_STRATEGY_RANDOM, nil

	default:
		return nil, fmt.Errorf("unknown coin selection strategy "+
			"%v", strategy)
	}
}
```

Reply:
you can just return `0` right? since it is an int under the hood and you are also returning an error and so the value wont be used

Reply:
Yes makes sense. Thanks!